File: windows/MDreader/Store/DocInfo.cs
namespace MDreader.Store;

/// <summary>
/// An immutable value snapshot of a cached document, decoupled from the SQLite
/// row so the UI holds plain data. Port of linux doc_info.rs / macOS DocInfo.swift.
/// </summary>
public sealed record DocInfo(
    Guid Id,
    string Title,
    string ContentHash,
    string? SourceUri,
    long OpenedAt,
    bool Favorite,
    long CharCount);

File: tui/src/store/doc_info.rs
// DocInfo — port of macOS DocInfo.swift. An immutable value snapshot of a cached document,
// decoupled from the SQLite row so the UI holds plain data.

use uuid::Uuid;

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct DocInfo {
    pub id: Uuid,
    pub title: String,
    pub content_hash: String,
    pub source_uri: Option<String>,
    pub opened_at: i64, // epoch millis
    pub favorite: bool,
    pub char_count: i64,
}
